src/Data/Loader/PostObjectLoader.php

```diff
@@ -93,70 +93,73 @@ function( $split, \WP_Query $query ) {
 				$loaded_posts[ $key ] = null;
 			}
 
-			/**
-			 * If there's a Post Author connected to the post, we need to resolve the
-			 * user as it gets set in the globals via `setup_post_data()` and doing it this way
-			 * will batch the loading so when `setup_post_data()` is called the user
-			 * is already in the cache.
-			 */
-			$context     = $this->context;
-			$user_id     = null;
-			$post_parent = null;
-
-			if ( ! empty( $post_object->post_author ) && absint( $post_object->post_author ) ) {
-
-				if ( ! empty( $post_object->post_author ) ) {
-					$user_id = $post_object->post_author;
-					$this->context->get_loader( 'user' )->buffer( [ $user_id ] );
-				}
-			}
-
-			if ( 'revision' === $post_object->post_type && ! empty( $post_object->post_parent ) && absint( $post_object->post_parent ) ) {
-				$post_parent = $post_object->post_parent;
-				$this->context->get_loader( 'post' )->buffer( [ $post_parent ] );
-			}
-
-			/**
-			 * This is a deferred function that allows us to do batch loading
-			 * of dependant resources. When the Model Layer attempts to determine
-			 * access control of a Post, it needs to know the owner of it, and
-			 * if it's a revision, it needs the Parent.
-			 *
-			 * This deferred function allows for the objects to be loaded all at once
-			 * instead of loading once per entity, thus reducing the n+1 problem.
-			 */
-			$load_dependencies = new Deferred(
-				function() use ( $post_object, $user_id, $post_parent, $context ) {
-
-					if ( ! empty( $user_id ) ) {
-						$context->get_loader( 'user' )->load( $user_id );
-					}
-					if ( ! empty( $post_parent ) ) {
-						$context->get_loader( 'post' )->load( $post_parent );
-					}
-
-					/**
-					 * Run an action when the dependencies are being loaded for
-					 * Post Objects
-					 */
-					do_action( 'graphql_post_object_loader_load_dependencies', $this, $post_object );
-
-					return;
-				}
-			);
-
-			/**
-			 * Once dependencies are loaded, return the Post Object
-			 */
-			$loaded_posts[ $key ] = $load_dependencies->then(
-				function() use ( $post_object ) {
-					$post = new Post( $post_object );
-					if ( ! isset( $post->fields ) || empty( $post->fields ) ) {
-						return null;
-					}
-					return $post;
-				}
-			);
+            if (!empty($post_object)) {
```

Comment:
@rburgst we have a check above that sets the `$loaded[ $key ] = null`. 

Perhaps we just change that to an if/else statement. 

```php
if ( ! $post_object instanceof \WP_Post ) {
   $loaded_posts[ $key ] = null;
} else {
   // all the other stuff here
}
```

Reply:
good point, done

Reply:
Thanks!